src/connection.h

```diff
@@ -79,13 +93,7 @@ typedef struct connection_t {
 	struct edge_t *edge;            /* edge associated with this connection */
 
 #ifndef DISABLE_LEGACY
-	rsa_t *rsa;                     /* his public RSA key */
-	cipher_t incipher;              /* Cipher he will use to send data to us */
-	cipher_t outcipher;             /* Cipher we will use to send data to him */
-	digest_t indigest;
-	digest_t outdigest;
-	uint64_t inbudget;
-	uint64_t outbudget;
+	legacy_ctx_t legacy;
```

Comment:
Yes, but the idea was also to make this a pointer, so we can opt not to allocate memory for it if we are not using the legacy protocol to communicate with the peer.

Reply:
Right, this makes more sense.

On an unrelated note, should I send in changes like [this one]?

[types]: https://github.com/hg/tinc/commit/5697f403b900989e68d9b0309ff0a54341ea9dda

Strictly speaking, it's quite useless since C type system is so primitive, but it at least makes intent more clear.

Reply:
Sure, I will merge patches like [that one](https://github.com/hg/tinc/commit/5697f403b900989e68d9b0309ff0a54341ea9dda). Although I wouldn't recommend spending too much time one things like this if you think there is more important stuff to do :)

Reply:
Sure, I don't spend time specifically on such stuff (there's not very much of it already). I replaced a couple of nids while working on this PR, thinking it will involve only one or two lines, but it turned out to be quite a bit larger than that.

Let's just throw that commit on top of this PR then, since it depends on it (unless it gets in the way).